File: HesapMakinesi.cs
using System;

public class HesapMakinesi
{
    public static void Main(string[] args)
    {
        double deger1 = 0, deger2 = 0, sonuc;
        int secim;

        while (true)
        {
            Console.WriteLine("\n**********Hesap Makinesi***********");
            Console.WriteLine("----------------");
            Console.WriteLine("1. Toplama");
            Console.WriteLine("2. Cikarma");
            Console.WriteLine("3. Carpma");
            Console.WriteLine("4. Bolme");
            Console.WriteLine("5. Kare Alma");
            Console.WriteLine("6. Karekok Hesaplama");
            Console.WriteLine("7. Cikis");

            Console.Write("Seciminizi girin (1-7): ");

            if (!int.TryParse(Console.ReadLine(), out secim))
            {
                Console.WriteLine("Gecersiz giris. Lutfen bir sayi girin.");
                continue; // Döngünün başına döner
            }

            if (secim == 7)
            {
                Console.WriteLine("Cikis yapiliyor...");
                break; // Döngüden çıkar
            }

            if (secim < 1 || secim > 6)
            {
                Console.WriteLine("Gecersiz secim. Lutfen 1-7 arasinda bir sayi girin.");
                continue;
            }

            Console.Write("Birinci sayiyi girin: ");
            if (!double.TryParse(Console.ReadLine(), out deger1))
            {
                Console.WriteLine("Gecersiz sayi girisi.");
                continue;
            }

            if (secim != 5 && secim != 6) // Kare alma ve karekök için ikinci sayı gerekmez
            {
                Console.Write("Ikinci sayiyi girin: ");
                if (!double.TryParse(Console.ReadLine(), out deger2))
                {
                    Console.WriteLine("Gecersiz sayi girisi.");
                    continue;
                }
            }

            switch (secim)
            {
                case 1:
                    sonuc = deger1 + deger2;
                    Console.WriteLine($"{deger1} + {deger2} = {sonuc}");
                    break;
                case 2:
                    sonuc = deger1 - deger2;
                    Console.WriteLine($"{deger1} - {deger2} = {sonuc}");
                    break;
                case 3:
                    sonuc = deger1 * deger2;
                    Console.WriteLine($"{deger1} * {deger2} = {sonuc}");
                    break;
                case 4:
                    if (deger2 == 0)
                    {
                        Console.WriteLine("0'a bolme hatasi!");
                    }
                    else
                    {
                        sonuc = deger1 / deger2;
                        Console.WriteLine($"{deger1} / {deger2} = {sonuc}");
                    }
                    break;
                case 5:
                    sonuc = Math.Pow(deger1, 2);
                    Console.WriteLine($"{deger1}'in karesi = {sonuc}");
                    break;
                case 6:
                    if (deger1 < 0)
                    {
                        Console.WriteLine("Negatif sayinin karekoku hesaplanamaz!");
                    }
                    else
                    {
                        sonuc = Math.Sqrt(deger1);
                        Console.WriteLine($"{deger1}'in karekoku = {sonuc}");
                    }
                    break;

            }
        }
    }
}

File: HesapMakinesi.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h> // math.h kütüphanesi eklendi

int main() {
    double sayi1, sayi2, sonuc;
    int secim;

    while (1) { // Sonsuz döngü
        printf("\nHesap Makinesi\n");
        printf("--------------\n");
        printf("1. Toplama\n");
        printf("2. Cikarma\n");
        printf("3. Carpma\n");
        printf("4. Bolme\n");
        printf("5. Kare Alma\n");
        printf("6. Karekok Hesaplama\n");
        printf("7. Cikis\n");

        printf("Seciminizi girin (1-7): ");

        if (scanf("%d", &secim) != 1) {
            printf("Gecersiz giris. Lutfen bir sayi girin.\n");
            while (getchar() != '\n'); // Giriş buffer'ını temizle
            continue;
        }

        if (secim == 7) {
            printf("Cikis yapiliyor...\n");
            break;
        }

        if (secim < 1 || secim > 6) {
            printf("Gecersiz secim. Lutfen 1-7 arasinda bir sayi girin.\n");
            continue;
        }

        printf("Birinci sayiyi girin: ");
        if (scanf("%lf", &sayi1) != 1) { // %lf double için
            printf("Gecersiz sayi girisi.\n");
            while (getchar() != '\n');
            continue;
        }

        if (secim != 5 && secim != 6) {
            printf("Ikinci sayiyi girin: ");
            if (scanf("%lf", &sayi2) != 1) {
                printf("Gecersiz sayi girisi.\n");
                while (getchar() != '\n');
                continue;
            }
        }

        switch (secim) {
            case 1:
                sonuc = sayi1 + sayi2;
                printf("%.2lf + %.2lf = %.2lf\n", sayi1, sayi2, sonuc);
                break;
            case 2:
                sonuc = sayi1 - sayi2;
                printf("%.2lf - %.2lf = %.2lf\n", sayi1, sayi2, sonuc);
                break;
            case 3:
                sonuc = sayi1 * sayi2;
                printf("%.2lf * %.2lf = %.2lf\n", sayi1, sayi2, sonuc);
                break;
            case 4:
                if (sayi2 == 0) {
                    printf("0'a bolme hatasi!\n");
                } else {
                    sonuc = sayi1 / sayi2;
                    printf("%.2lf / %.2lf = %.2lf\n", sayi1, sayi2, sonuc);
                }
                break;
            case 5:
                sonuc = pow(sayi1, 2); // pow fonksiyonu
                printf("%.2lf'nin karesi = %.2lf\n", sayi1, sonuc);
                break;
            case 6:
                if (sayi1 < 0) {
                    printf("Negatif sayinin karekoku hesaplanamaz!\n");
                } else {
                    sonuc = sqrt(sayi1); // sqrt fonksiyonu
                    printf("%.2lf'nin karekoku = %.2lf\n", sayi1, sonuc);
                }
                break;
            default:
                printf("Gecersiz secim.\n"); // Gerekli değil ama eklendi
        }
    }

    return 0;
}